packages/library-legacy/src/connection.ts

```diff
@@ -3724,7 +3724,17 @@ export class Connection {
     commitment?: Commitment,
   ): Promise<RpcResponseAndContext<SignatureResult>>;
 
-  /** @deprecated Instead, call `confirmTransaction` and pass in {@link TransactionConfirmationStrategy} */
+  /** @deprecated Instead, call `confirmTransaction` and pass in {@link TransactionConfirmationStrategy}
+   *  @example
+   * ``` ts
+   * const latestBlockHash = await connection.getLatestBlockhash()
+   * await connection.confirmTransaction({
+   *   blockhash: latestBlockHash.blockhash,
+   *    lastValidBlockHeight: latestBlockHash.lastValidBlockHeight,
+   *    signature: airdropSignature,
+   *   })
+   *   ```
```

Comment:
If we're going to include an example, I think it should be more explicit.

Such a usage description should be detailing exactly how to set up the confirmation strategy. We can just demonstrate the blockheight-based one for this doc.

The meat of the blockheight strategy is that the provided `blockhash` and `lastValidBlockheight` are used to constrain the range in which the transaction should be confirmed. Typically this is going to be the same values returned by `getLatestBlockhash` when the transaction is constructed, but for some use cases it may make sense to use an older blockhash.

For the sake of this example, I'd recommend something like this:

```ts
const { blockhash, lastValidBlockheight } = await connection.getLatestBlockhash();

const tx = new Transaction({ blockhash, lastValidBlockheight });
const signature = await connection.sendTransaction(tx);

const confirmationStrategy: BlockheightBasedTransactionConfirmationStrategy = {
  blockhash,
  lastValidBlockheight,
  signature,
};
await connection.confirmTransaction(confirmationStrategy);
```

But we should make sure to document that this is specifically for blockheight confirmation. Ideally we'd also want an accompanying nonce strategy example as well.

Reply:
I agree with this. Should we add it to this PR or make a separate one? (as mentioned in the other comment, the intention was to help people that want to use the deprecated function, that is still mentioned in guides, tutorials, SA answers)

Reply:
Okay I added the example on the non deprecated `confirmTransaction` method. 
I also found a different spot where an example used the old signature. Updated that as well: https://github.com/solana-labs/solana-web3.js/pull/2733/commits/d672f34539b7d5f155268ec9f843a5d06ac4a645

Regarding the DurableNonce example: 
In most cases a blockhash based confirmation strategy will be used. I hinted at the possibility of a nonce based one next to the link  to the `TransactionConfirmationStrategy` type. It could be a good spot to put the example there. So anyone who is curious will click and everyone else can move forward.